codec-http2/src/main/java/io/netty/handler/codec/http2/UniformStreamByteDistributor.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.netty.handler.codec.http2;
+
+import static io.netty.handler.codec.http2.Http2Error.INTERNAL_ERROR;
+import static io.netty.handler.codec.http2.Http2Exception.connectionError;
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+import static java.lang.Math.max;
+import static java.lang.Math.min;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+/**
+ * A {@link StreamByteDistributor} that ignores stream priority and uniformly allocates bytes to all
+ * streams. This class uses a minimum chunk size that will be allocated to each stream. While
+ * fewer streams may be written to in each call to {@link #distribute(int, Writer)}, doing this
+ * should improve the goodput on each written stream.
+ */
+public final class UniformStreamByteDistributor implements StreamByteDistributor {
+    static final int DEFAULT_MIN_ALLOCATION_CHUNK = 1024;
+
+    private final Http2Connection.PropertyKey stateKey;
+    private final Deque<State> queue = new ArrayDeque<State>(4);
```

Comment:
This queue will only ever be as big as the number of active streams. So the same would be true for `DefaultHttp2Connection`.  I don't think we should handle this structure differently than other parts of the code.  If we don't like this, we should address it separately ... and apply the change across the codebase.  WDYT?